xcodecommand/xcpretty.go

```diff
@@ -59,38 +63,54 @@ func (c *xcprettyCommandRunner) Run(workDir string, xcodebuildArgs []string, xcp
 		}
 	}()
 
-	c.logger.TPrintf("$ set -o pipefail && %s | %v", buildCmd.PrintableCommandArgs(), prettyCmd.PrintableCommandArgs())
+	c.logger.TPrintf("$ set -o pipefail && %s | %s", buildCmd.PrintableCommandArgs(), prettyCmd.PrintableCommandArgs())
 
-	if err := buildCmd.Start(); err != nil {
-		return Output{
-			RawOut:   buildOutBuffer.Bytes(),
-			ExitCode: 1,
-		}, err
+	err := buildCmd.Start()
+	if err == nil {
+		err = prettyCmd.Start()
 	}
-	if err := prettyCmd.Start(); err != nil {
-		return Output{
-			RawOut:   buildOutBuffer.Bytes(),
-			ExitCode: 1,
-		}, err
+	if err == nil {
+		err = buildCmd.Wait()
 	}
 
-	if err := buildCmd.Wait(); err != nil {
+	exitCode := 0
+	if err != nil {
+		exitCode = -1
+
 		var exerr *exec.ExitError
 		if errors.As(err, &exerr) {
-			return Output{
-				RawOut:   buildOutBuffer.Bytes(),
-				ExitCode: exerr.ExitCode(),
-			}, err
+			exitCode = exerr.ExitCode()
 		}
-
-		return Output{
-			RawOut:   buildOutBuffer.Bytes(),
-			ExitCode: 1,
-		}, err
 	}
 
 	return Output{
 		RawOut:   buildOutBuffer.Bytes(),
-		ExitCode: 0,
-	}, nil
+		ExitCode: exitCode,
+	}, err
+}
+
+func (c *xcprettyCommandRunner) cleanOutputFile(xcprettyArgs []string) {
```

Comment:
I can see why this is needed unfortunately, but why is this part of the PR, why did this only become an issue now?
I feel that this is a workaround and the better approach would be to use unique output paths (and then copy them over to the deploy dir), but I see that the path is hardcoded into the default value of the input.

Reply:
Nevermind, I see that this is moved over from a util file and this workaround exists in the current step too 🤷